### PR TITLE
chore(s3): upload should guess content type as possible

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/aws/s3/s3.go [hunk 1]
```diff
@@ -218,6 +221,15 @@ func (s *S3) upload(bucket, key string, buf io.Reader) (string, error) {
 	return resp.Location, nil
 }
 
+func defaultContentTypeFromExt(key string) *string {
+	contentType := mime.TypeByExtension(filepath.Ext(key))
+	if contentType == "" {
+		return nil
```
Comment (Lou1415926) [inline internal/pkg/aws/s3/s3.go:227]: aye just curious - do you know if s3 tries to determine the content type when it's ""? Or does it just default to something.

File: internal/pkg/aws/s3/s3.go [hunk 2]
```diff
@@ -218,6 +221,15 @@ func (s *S3) upload(bucket, key string, buf io.Reader) (string, error) {
...
+	}
+	fmt.Println(contentType)
```
Comment (Lou1415926) [inline internal/pkg/aws/s3/s3.go]: :doge Just to be annoying and point out trivial things

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/aws/s3/s3.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/aws/s3/s3.go:227]: i don't know about an empty string but if it is nil s3 does have a default value (which mostly doesn't make any sense and the link will download the object)

File: internal/pkg/aws/s3/s3.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/aws/s3/s3.go]: :homer-disappearing:

**Review by huanjani — APPROVED** [review]
This is cool!

**Review by efekarakus — COMMENTED** [review]
noice

**Pull request merged**